src/hotspot/share/opto/superword.hpp

```diff
@@ -657,9 +657,16 @@ class SWPointer : public ArenaObj {
   bool        _analyze_only; // Used in loop unrolling only for swpointer trace
   uint        _stack_idx;    // Used in loop unrolling only for swpointer trace
 
-  PhaseIdealLoop* phase() const { return _slp->phase(); }
-  IdealLoopTree*  lpt() const   { return _slp->lpt(); }
-  PhiNode*        iv() const    { return _slp->iv();  } // Induction var
+  PhaseIdealLoop* _phase;    // PhaseIdealLoop handle
+  IdealLoopTree*  _lpt;      // Current IdealLoopTree
+
+  PhaseIdealLoop* phase() const { return _phase; }
+  IdealLoopTree*  lpt() const   { return _lpt; }
+  PhiNode* iv() const {
+    return _slp ? _slp->iv() : _lpt->_head->as_CountedLoop()->phi()->as_Phi();
```

Comment:
I'd suggest either cache it directly from `_lpt->_head->as_CountedLoop()->phi()->as_Phi()`, or just query it directly. Reduce dependence on `_slp`.

Reply:
I think that would be optimal, if it is possible. I would maybe call it a `CLPointer`, for counted-loop-pointer? And only have a reference to the `_lpt` / `cl`. Eventually, we may want to even allow non-conted-loops, but that is really for the future.

Reply:
Done in SWPointer refactoring patch